go/vt/vtgate/vcursor_impl.go

```diff
@@ -1428,6 +1433,7 @@ func (vc *vcursorImpl) CloneForMirroring(ctx context.Context) engine.VCursor {
 		warnShardedOnly:     vc.warnShardedOnly,
 		warnings:            vc.warnings,
 		pv:                  vc.pv,
+		resultsObserver:     vc.resultsObserver,
```

Comment:
should this have been a `nullResultsObserver{}` instead?

Reply:
I guess I'd probably want to not observe results from a mirrored version of the query for the same reasons that we create a new LogStats (i.e. combining the queries could be confusing). I'll update this to be `nullResultsObserver` unless you think otherwise.